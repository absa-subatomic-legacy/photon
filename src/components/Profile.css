.hello {
  text-align: center;
  margin: 20px;
  font-size: 16;
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
}

.hello button {
  margin-left: 25px;
  margin-right: 25px;
  font-size: 16;
  min-width: 50px;
}
