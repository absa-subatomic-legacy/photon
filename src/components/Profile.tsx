import * as React from 'react';

import './Profile.css';

export interface Props {
  name: string;
  forkProject?: (url: string) => void;
}

export default function Profile({ name, forkProject }: Props) {
  var click = () => {
    if (forkProject) {
      forkProject('https://github.com/absa-subatomic/photon');
    }
  };
  
  return (
    <div>
      <p>
        Hello! My name is <strong>{name}</strong>.
        My name was passed in through a route param!
      </p>
      <button onClick={click}>Fork the Project</button>
    </div>
  );
}
