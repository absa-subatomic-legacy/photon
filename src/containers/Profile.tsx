import Profile from '../components/Profile';
import * as actions from '../actions';
import { ProjectState } from '../types';
import { connect, Dispatch } from 'react-redux';

// tslint:disable-next-line: no-any
function mapStateToProps(state: {projectState: ProjectState}, ownProps: any) {
    return {
        name: ownProps && ownProps.match ? ownProps.match.params.name : '',
    };
}

function mapDispatchToProps(dispatch: Dispatch<actions.ProjectAction>) {
    return {
        forkProject: (url: string) => {
            return actions.forkProject(url)(dispatch);
        },
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Profile);