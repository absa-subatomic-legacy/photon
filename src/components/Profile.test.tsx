import * as React from 'react';
import { shallow } from "enzyme";
import Profile from './Profile';

describe('A Profile', function() {
  it('renders the correct text when no enthusiasm level is given', () => {
    const hello = shallow(<Profile name="Daniel" />);
    expect(hello.find('.greeting').text()).toEqual('Hello Daniel!');
  });

  it('renders the correct text with an explicit enthusiasm of 1', () => {
    const hello = shallow(<Profile name="Daniel" enthusiasmLevel={1}/>);
    expect(hello.find('.greeting').text()).toEqual('Hello Daniel!');
  });

  it('renders the correct text with an explicit enthusiasm level of 5', () => {
    const hello = shallow(<Profile name="Daniel" enthusiasmLevel={5} />);
    expect(hello.find('.greeting').text()).toEqual('Hello Daniel!!!!!');
  });

  it('throws when the enthusiasm level is 0', () => {
    expect(() => {
      shallow(<Profile name="Daniel" enthusiasmLevel={0} />);
    }).toThrow();
  });

  it('throws when the enthusiasm level is negative', () => {
    expect(() => {
      shallow(<Profile name="Daniel" enthusiasmLevel={-1} />);
    }).toThrow();
  });
});