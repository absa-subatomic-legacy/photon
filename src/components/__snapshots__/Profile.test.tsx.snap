// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Snapshot of Profile Component capturing Snapshot of Profile 1`] = `
<div
  className="hello"
>
  <p
    className="greeting"
  >
    Hello! My name is 
    <strong>
      Daniel
    </strong>
    . My name was passed in through a route param!
  </p>
  <button
    onClick={[Function]}
  >
    Fork the Project
  </button>
</div>
`;
